jetty-http/src/main/java/org/eclipse/jetty/http/pathmap/PathMappings.java

```diff
@@ -46,7 +46,7 @@
 public class PathMappings<E> implements Iterable<MappedResource<E>>, Dumpable
 {
     private static final Logger LOG = LoggerFactory.getLogger(PathMappings.class);
-    private final Set<MappedResource<E>> _mappings = new TreeSet<>();
+    private final Set<MappedResource<E>> _mappings = new TreeSet<>(LogicalDeclarationComparator.INSTANCE);
```

Comment:
Rather than have another class and file, why not:
```suggestion
    private final Set<MappedResource<E>> _mappings = new TreeSet<>((a,b)->a.getPathSpec().compareTo(b.getPathSpec());
```